### PR TITLE
Feature/cas on boarding tactical solution v1

### PR DESCRIPTION
**Pull request by VijayHirudayasamy-bc**

### CODE REVIEW

File: api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+      {
+        roleList = roles.ToList();
+      }
+
+
+      var defaultRoles = await _dataContext.CcsAccessRole.Where(ar => !ar.IsDeleted && roleList.Contains(ar.CcsAccessRoleNameKey)).ToListAsync();
+
+      List<OrganisationEligibleRole> addedEligibleRoles = new List<OrganisationEligibleRole>();
+
+      var orgRoles = await _dataContext.OrganisationEligibleRole.Where(x => x.OrganisationId == orgId).ToListAsync();
```
Comment (brijrajsinh-bc) [inline api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs:261]: isdeleted condition missing

File: api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+      {
+        _dataContext.OrganisationEligibleRole.AddRange(addedEligibleRoles);
+        await _dataContext.SaveChangesAsync();
+      }
+
+
+
+      foreach (var userName in adminList)
+      {
+        var user = await _dataContext.User
```
Comment (brijrajsinh-bc) [inline api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs:292]: primary user condition missing

File: api/CcsSso.Core.ServiceOnboardingScheduler/Service/CASOnBoardingService.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    public async Task<List<Tuple<int, string, string, DateTime>>> GetOrgAdmins(int orgId, string ciiOrganisationId)
+    {
+
+      var orgAdminAccessRole = await _dataContext.OrganisationEligibleRole.FirstOrDefaultAsync(or => !or.IsDeleted && or.OrganisationId == orgId && or.CcsAccessRole.CcsAccessRoleNameKey == Contstant.OrgAdminRoleNameKey);
+
+      if (orgAdminAccessRole != null)
+      {
+        var users = await _dataContext.User
```
Comment (brijrajsinh-bc) [inline api/CcsSso.Core.ServiceOnboardingScheduler/Service/CASOnBoardingService.cs:92]: verified condition missing.

File: api/CcsSso.Core.ServiceOnboardingScheduler/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft": "Error",
+      "Microsoft.Hosting.Lifetime": "Error"
+    }
+  },
+  "VaultEnabled": false,
```
Comment (brijrajsinh-bc) [inline api/CcsSso.Core.ServiceOnboardingScheduler/appsettings.json:9]: vaultenable value should be committed as true

File: api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+          _logger.LogInformation($"No user found in user table for user {userName}");
+          continue;
+        }
+
+        List<UserAccessRole> userAccessRoles = new List<UserAccessRole>();
+
+        foreach (var role in defaultRoles)
+        {
+          var organisationAdminAccessRole = await _dataContext.OrganisationEligibleRole
+           .FirstOrDefaultAsync(oer => !oer.IsDeleted && oer.OrganisationId == user.Party.Person.OrganisationId && oer.CcsAccessRole.CcsAccessRoleNameKey == role.CcsAccessRoleNameKey && oer.OrganisationId == orgId);
```
Comment (brijrajsinh-bc) [inline api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs:308]: You are already getting org id as param no need to take using users i think. Also you are already making call to OrganisationEligibleRole in above code. we can reduce db call.

File: api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs (hunk 2 above)
Comment (VijayHirudayasamy-bc) [reply api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs:293]: added

File: api/CcsSso.Core.ServiceOnboardingScheduler/appsettings.json (hunk 4 above)
Comment (VijayHirudayasamy-bc) [reply api/CcsSso.Core.ServiceOnboardingScheduler/appsettings.json:9]: Only using app secrets

**Review by brijrajsinh-bc — APPROVED** [review]

**Pull request merged**